### ISSUE TITLE
Output of tf.random.uniform can equal maxval, contrary to docstring

### ISSUE THREAD

**Issue by NoahDStein** [tensorflow/tensorflow#51465]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS Big Sur 11.2.3 (also observed on Mojave 10.14.6 and on Amazon Linux 2)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.6.0 (also observed on 2.4.1)
- Python version: 3.9.0 (also observed on 3.7 and 3.9.6)
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A (observed on CPU)
- GPU model and memory: N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
The function tf.random.uniform sometimes returns values equal to `maxval`, as in the code below.

**Describe the expected behavior**
According to the docstring for `tf.random.uniform`, "The generated values follow a uniform distribution in the range [minval, maxval). The lower bound `minval` is included in the range, while the upper bound `maxval` is excluded."

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): no
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Python 3.9.0 (default, Nov 27 2020, 21:03:12)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.19.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import tensorflow as tf

In [2]: tf.random.set_seed(42)

In [3]: tf.random.uniform((10000000,), minval=5, maxval=6, seed=42)[1739846]
2021-08-12 13:20:28.540136: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Out[3]: <tf.Tensor: shape=(), dtype=float32, numpy=6.0>

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by sanatmpa1** [comment]
@jvishnuvardhan,

I am able to reproduce the issue in `TF 2.5.0, TF 2.6.0 and tf-nightly` which indicates the presence of upper bound `maxval` in generated values, and please find the [gist here](https://colab.research.google.com/gist/sanatmpa1/a0a7ac88fcbfb016b3bcc2b6caea4912/51465.ipynb). Thanks!

**Comment by jvishnuvardhan** [comment]
@NoahDStein Looks like this is a bug. 

When I tried with `numpy`, `maxval` is excluded from the distribution. But, `maxval` is included (as opposed to what mentioned in the docs) in TF. We will look into root-cause of the issue. Thanks!

**Comment by PaulKlinger** [comment]
@NoahDStein @jvishnuvardhan This is the same behaviour as in numpy, see the comment in [the numpy docs there](https://numpy.org/doc/stable/reference/random/generated/numpy.random.uniform.html): 

> The high limit may be included in the returned array of floats due to floating-point rounding in the equation low + (high-low) * random_sample().

Here the output of 
```Python
tf.random.uniform((10000000,), seed=42)[1739846]
```
is `0.99999976`, and, as float precision drops away from zero,
```
np.float32(5) + np.float32(0.99999976) == 6
````

It might be helpful to add a similar comment in the tensorflow docs?

**Comment by mohantym** [comment]
Hi @NoahDStein !
It is working as expected if we add a multiplication operator (*) before [1739846] . 
Attached few test case in the attached [gist](https://colab.sandbox.google.com/gist/mohantym/249b5d2d1e303f442be564253190140e/git_51465.ipynb) for your reference.

Thank you!
